external-crates/move/crates/move-bytecode-verifier/src/regex_reference_safety/mod.rs

```diff
@@ -0,0 +1,547 @@
+
+        Bytecode::BrTrue(_) | Bytecode::BrFalse(_) => {
+            safe_assert!(safe_unwrap_err!(verifier.stack.pop()).is_non_ref());
+        }
+
+        Bytecode::Abort => {
+            safe_assert!(safe_unwrap_err!(verifier.stack.pop()).is_non_ref());
+            state.abort()
+        }
+        Bytecode::LdTrue
+        | Bytecode::LdFalse
+        | Bytecode::LdU8(_)
+        | Bytecode::LdU16(_)
+        | Bytecode::LdU32(_)
+        | Bytecode::LdU64(_)
+        | Bytecode::LdU128(_)
+        | Bytecode::LdU256(_)
+        | Bytecode::LdConst(_) => verifier.push(AbstractValue::NonReference)?,
+
+        Bytecode::Add
+        | Bytecode::Sub
+        | Bytecode::Mul
+        | Bytecode::Mod
+        | Bytecode::Div
+        | Bytecode::BitOr
+        | Bytecode::BitAnd
+        | Bytecode::Xor
+        | Bytecode::Shl
+        | Bytecode::Shr
+        | Bytecode::Or
+        | Bytecode::And
+        | Bytecode::Lt
+        | Bytecode::Gt
+        | Bytecode::Le
+        | Bytecode::Ge => {
+            safe_assert!(safe_unwrap_err!(verifier.stack.pop()).is_non_ref());
+            safe_assert!(safe_unwrap_err!(verifier.stack.pop()).is_non_ref());
+            // TODO maybe call state.value_for
+            verifier.push(AbstractValue::NonReference)?
+        }
+
+        Bytecode::Pack(idx) => {
+            let struct_def = verifier.module.struct_def_at(*idx);
+            pack_struct(verifier, struct_def)?
+        }
+        Bytecode::PackGeneric(idx) => {
+            let struct_inst = verifier.module.struct_instantiation_at(*idx);
+            let struct_def = verifier.module.struct_def_at(struct_inst.def);
+            pack_struct(verifier, struct_def)?
+        }
+        Bytecode::Unpack(idx) => {
+            let struct_def = verifier.module.struct_def_at(*idx);
+            unpack_struct(verifier, struct_def)?
+        }
+        Bytecode::UnpackGeneric(idx) => {
+            let struct_inst = verifier.module.struct_instantiation_at(*idx);
+            let struct_def = verifier.module.struct_def_at(struct_inst.def);
+            unpack_struct(verifier, struct_def)?
+        }
+
+        Bytecode::VecPack(_, num) => {
+            if let Some(num_to_pop) = NonZeroU64::new(*num) {
+                let result = verifier.stack.pop_eq_n(num_to_pop);
+                let abs_value = safe_unwrap_err!(result);
+                safe_assert!(abs_value.is_non_ref());
+            }
+
+            verifier.push(AbstractValue::NonReference)?;
+        }
+
+        Bytecode::VecLen(_) => {
+            let vec_ref = safe_unwrap_err!(verifier.stack.pop());
+            state.vector_op(offset, vec_ref, false, meter)?;
+            verifier.push(AbstractValue::NonReference)?;
+        }
+
+        Bytecode::VecImmBorrow(_) => {
+            safe_assert!(safe_unwrap_err!(verifier.stack.pop()).is_non_ref());
+            let vec_ref = safe_unwrap_err!(verifier.stack.pop());
+            let values = state.call(
```

Comment:
Maybe a stupid question, but is there a reason these are being modeled as calls when they are not?

Reply:
Because they behave like calls--they make a `.*`